_posts/2024-01-16-owasp-top10-with-abstract-examples-part-2.md

```diff
@@ -0,0 +1,120 @@
+
+Let's imagine that the order process works internally with pieces of paper. If the kitchen doesn't pay attention to the origin of the order papers, someone can slip the order under the kitchen door and write on the order that the food should be delivered through the back door, no questions asked. Another breach of integrity would be if the cashier blurted out all the restaurant's secrets to the customer, whether they asked or not.
+
+### The fix
+
+We should somehow try to raise the trust factor of our data and libraries. We could scan the products we use for common vulnerabilities and fix them when patches are available. We should also enforce digital signatures to ensure that updated information and code are only installed from trusted sources.
+
+In our example, shaking up the cash register and kitchen staff would be in place. They shouldn't rely on random order notes in the kitchen. especially if they are found in suspicious places like the floor. Also, the cashier should not leak / overflow information to a random customer, but stick strictly to taking orders.
+
+
+## 9 Security Logging and Monitoring Failures
+
+When we don't have a complete view of what is happening in our application, then we have a [logging and monitoring risk](https://owasp.org/Top10/A09_2021-Security_Logging_and_Monitoring_Failures/) at hand. Without good logging and monitoring, we use our software more or less blindly. How do we find out about unauthorized access or that our software is down due to a traffic spike? It's also pointless to produce errors that no one can use or understand.
+
+Let's assume that our restaurant would not properly monitor what is happening. They do not have cameras in the driving lane and the purchase transaction is monitored only with the minimum amount required by law. The restaurant would have great difficulty in monitoring which products are sold and developing its business accordingly. There would also be a lack of communication. For example, if the kitchen makes a mistake and prepares the wrong product, the cashier just shouts "ERROR" without saying what went wrong.
+
+### The fix
+
+When we use our services and applications, we should have a good view of how they work and who uses them with what rights. Tracking should be used to provide insight into the status of our services. Is it healthy, struggling or down. Logging can be also used to detect strange anomalies and suspicious user actions.
+
+![Logging and monitoring](/img/2024-01-owasp-top-10/9_logging.jpg)
+
+Surveillance cameras should be added to the driving lane to detect vandalism and abuse. Purchase transactions should also be recorded at a level that would enable the management of the company. The cash register should also increase communication with the kitchen so that possible errors are understood and corrective measures are implemented.
```

Comment:
"enable the management of the company"... to do what? vai tarkoittaako yleisesti, että se mahdollistaa yrityksen johtamisen

Reply:
Korjasin, että transaktiot kannattaa merkata tasolle, joka mahdollistaa päivittäisen busineksen pyörittämisen. Ei sinänsä firman ajamisen.